benchmarks/upload_metrics_to_bq.py

```diff
@@ -0,0 +1,186 @@
+
+# It is important to parse the last n steps avoiding steps that have tracing turned on, by default last_n_steps should be total_steps - 10 to avoid initial tracing settings.
+# TODO() Support avoiding tracing when the user explicitly enables tracing for other steps.
+def parse_metrics(local_metrics_file, total_steps, last_n_steps=10) -> Metrics:
+  avg_tflops = get_metric_average(local_metrics_file, "perf/per_device_tflops_per_sec", n=last_n_steps)
+  avg_tokens_per_second = get_metric_average(local_metrics_file, "perf/per_device_tokens_per_sec", n=last_n_steps)
+  median_step_time = get_metric_median(local_metrics_file, "perf/step_time_seconds", n=last_n_steps)
+  e2e_step_time = get_metrics_sum(local_metrics_file, "perf/step_time_seconds", n=total_steps)
+
+  metrics = Metrics(
+    avg_tflops_per_sec=avg_tflops,
+    avg_tokens_per_sec=avg_tokens_per_second,
+    median_step_time=median_step_time,
+    e2e_step_time=e2e_step_time)
+
+  return metrics
+
+# Args
+# metric gcs file location
+#
+def main(argv: Sequence[str]) -> None:
+  metrics_gcs_file = argv[0]
+  model_id = argv[1]
+  hardware_id = argv[2]
+  software_id = argv[3]
+  number_of_chips = argv[4]
+  container_image_name = argv[5]
+  global_batch_size = argv[6]
+  precision = argv[7]
+  optimizer = argv[8]
+  seq_length = argv[9]
+  number_of_steps = argv[10]
+  xla_flags = argv[11]
+  dataset = argv[12]
+
+  local_dir = DEFAULT_LOCAL_DIR
+
+  # 1.Download metrics from the GCS Bucket
+  local_metrics_file = local_dir
+  rc = download_metrics_file_locally(metrics_gcs_file=metrics_gcs_file, local_file=local_metrics_file)
+  if rc != 0:
+    exit(rc)
+
+  # 2.Install MantaRay locally
+  rc = install_mantaray_locally(local_dir)
+  if rc != 0:
+    exit(rc)
+
+  # 3.Parse Metrics
+  # If there are more than 10 steps, have a buffer to avoid profiling bad perf:
+  if number_of_steps - 10 > 0:
+    compute_metrics_of_n_steps = number_of_steps - 10
+  else:
+    compute_metrics_of_n_steps = number_of_steps
+
+  metrics_from_file = parse_metrics(local_metrics_file, number_of_steps, compute_metrics_of_n_steps)
+
+  # TODO() convert hardware_id to bq acceptable term
+  hardware_id = hardware_id
+  # TODO() convert number of chips to number of nodes (number of vms)
+  # Trillium 4 chips per vm
+  number_of_nodes = number_of_chips / 4
+
+  # TODO() convert tflops to MFU based on hardware_id
+  # Trillium 918
+  avg_mfu = metrics_from_file.avg_tflops_per_sec / 918
```

Comment:
@stingram  we should have this info per hardware_id, can we have utility functions to get this information?